app/submitter/submitter.py

```diff
@@ -141,3 +143,46 @@ def send_message(self, message, tx_id, questionnaire_id, case_id=None):
         finally:
             if connection:
                 self._disconnect(connection)
+
+
+class GCSFeedback:
+    def __init__(self, bucket_name):
+        client = storage.Client()
+        self.bucket = client.get_bucket(bucket_name)
+        self.bucket.retention_period = 86400
+
+    def upload(self, schema, feedback_data, session_data):
+        object_key = str(uuid4())
+        blob = self.bucket.blob(object_key)
+        blob.metadata = {
+            "feedback_count": session_data.feedback_count + 1,
```

Comment:
We shouldn't be incrementing in 2 places but this would be addressed by my comment about passing in data and metadata that needs to be uploaded instead of constructing the object here.

Reply:
This isn't an increment, session_data.feedback_count stays the same. I can't increment session_data.feedback_count until the upload is successfully, I can't see how its possible to know it that

Reply:
Based on my other comments this should change anyway as you would be able to create the object upfront and pass to upload but in this instance you could increment the session data which was passed in here and session data is only then saved if the upload is successful.

Reply:
You could separate the increment and the save, but that has future bug written all over it.

Reply:
It would be done in the same code block based on my changed.